api/tests/integration/UserTest.php

```diff
@@ -222,4 +217,21 @@ public function testDeleteOneByGuestUser()
         $this->assertResponseStatus(403);
         $this->assertEquals($rowCount, User::count());
     }
+
+    public function testResetPasswordMail()
+    {
+        $this->clearMessages();
+        $user = $this->createUser('guest');
+        $token = $this->tokenFromUser($user);
+
+        $this->delete('/users/'.$user->user_id.'/password', [], [
+            'HTTP_AUTHORIZATION' => 'Bearer '.$token
+        ]);
+
+        $mail = $this->getLastMessage();
+
+        $this->assertResponseStatus(202);
+        $this->assertResponseHasNoContent();
+        $this->assertContains('Password', $mail->subject);
+    }
```

Comment:
@artstorm specifically I mean extracting the token from the email to test, as when we update the emails to look pretty we want to have make sure this functionality is not broken. We may have to revisit how the test is written, but at least a developer will be looking at the tests.